### PR TITLE
Sold products page

### PR DESCRIPTION
**Pull request by JellyLu**

### CODE REVIEW

File: src/containers/pages/SoldProductsPage/SoldProductsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { connect, DispatchProp } from 'react-redux';
+import { RouteComponentProps } from 'react-router';
+import { goBack } from 'react-router-redux';
+
+import * as D from '../../../definitions';
+import { Header, Footer, Product } from '../../../components';
+import { soldProductsAction } from '../../../modules/products/actions';
+import './SoldProductsPage.css'
+
+type SoldProductPageProps<S> = DispatchProp<S> & RouteComponentProps<S> & {
```
Comment (demozhang1029) [inline src/containers/pages/SoldProductsPage/SoldProductsPage.tsx]: confuse about the <S>...It's the typescript grammar or an interface?

File: src/containers/pages/SoldProductsPage/SoldProductsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  _renderProductsList() {
+    const {soldProducts} = this.props;
+    return (
+      <div className="content">
+        {
+          soldProducts &&
+          soldProducts.map(product => {
+            const ob = {
```
Comment (shane123qaz) [inline src/containers/pages/SoldProductsPage/SoldProductsPage.tsx]: Can you give some meaningful name...

File: src/definitions.ts [hunk 3]
```diff
@@ -72,7 +72,8 @@ export type HomeProductsState = HomeProducts;
 export interface RootState<S> {
     user?: UserState;
     app?: AppState;
-    boughtProducts?: ProductsState;
+    boughtProducts?: ProductsState;g
```
Comment (shane123qaz) [inline src/definitions.ts]: g...

File: src/apis/product.ts [hunk 4]
```diff
@@ -31,4 +48,4 @@ export const homeProducts = (): Promise<D.Products> => {
     }
     return null;
```
Comment (shane123qaz) [inline src/apis/product.ts]: no return should has the same meaning with return null.

File: src/definitions.ts (hunk 3 above)
Comment (benben198805) [inline src/definitions.ts]: g?

File: src/containers/pages/SoldProductsPage/SoldProductsPage.tsx (hunk 2 above)
Comment (JellyLu) [reply src/containers/pages/SoldProductsPage/SoldProductsPage.tsx]: will do in next pr

**Pull request merged**